botocore/tokens.py

```diff
@@ -248,6 +249,14 @@ def _client(self):
         )
         return self._session.create_client("sso-oidc", config=config)
 
+    @CachedProperty
+    def profile_name(self):
```

Comment:
Let's avoid making this a public property on the class. In general, this token class needs to follow the interface of the base token class so that we can uphold the property that all token classes has a specific interface, which in the end simplifies downstream usages. Instead, I'd recommend just computing the profile name and setting it to `self._profile_name` as part of the initializer.